--- a/Makefile
+++ b/Makefile
@@ -164,11 +164,11 @@ endif
 include $(srcdir)/Makefile.include
 
 
-LIBMCOUNT_TARGETS := libmcount/libmcount.so libmcount/libmcount-fast.so
-LIBMCOUNT_TARGETS += libmcount/libmcount-single.so libmcount/libmcount-fast-single.so
+LIBMCOUNT_TARGETS := libmcount/libmcount.a libmcount/libmcount-fast.a
+LIBMCOUNT_TARGETS += libmcount/libmcount-single.a libmcount/libmcount-fast-single.a
 
-_TARGETS := uftrace libtraceevent/libtraceevent.a python/uftrace_python.so
-_TARGETS += $(LIBMCOUNT_TARGETS) libmcount/libmcount-nop.so
+_TARGETS := uftrace libtraceevent/libtraceevent.a python/uftrace_python.a
+_TARGETS += $(LIBMCOUNT_TARGETS) libmcount/libmcount-nop.a
 _TARGETS += misc/demangler misc/symbols misc/dbginfo
 TARGETS  := $(patsubst %,$(objdir)/%,$(_TARGETS))
 
@@ -280,20 +280,20 @@ $(LIBMCOUNT_FAST_SINGLE_OBJS): $(objdir)/%-fast-single.op: $(srcdir)/%.c $(LIBMC
 $(LIBMCOUNT_NOP_OBJS): $(objdir)/%.op: $(srcdir)/%.c $(LIBMCOUNT_DEPS)
 	$(QUIET_CC_FPIC)$(CC) $(LIB_CFLAGS) -c -o $@ $<
 
-$(objdir)/libmcount/libmcount.so: $(LIBMCOUNT_OBJS) $(LIBMCOUNT_UTILS_OBJS) $(LIBMCOUNT_ARCH_OBJS)
-	$(QUIET_LINK)$(CC) -shared -o $@ $^ $(LIB_LDFLAGS)
+$(objdir)/libmcount/libmcount.a: $(LIBMCOUNT_OBJS) $(LIBMCOUNT_UTILS_OBJS) $(LIBMCOUNT_ARCH_OBJS)
+	$(QUIET_LINK)ar rcs $@ $^
 
-$(objdir)/libmcount/libmcount-fast.so: $(LIBMCOUNT_FAST_OBJS) $(LIBMCOUNT_UTILS_OBJS) $(LIBMCOUNT_ARCH_OBJS)
-	$(QUIET_LINK)$(CC) -shared -o $@ $^ $(LIB_LDFLAGS)
+$(objdir)/libmcount/libmcount-fast.a: $(LIBMCOUNT_FAST_OBJS) $(LIBMCOUNT_UTILS_OBJS) $(LIBMCOUNT_ARCH_OBJS)
+	$(QUIET_LINK)ar rcs $@ $^
 
-$(objdir)/libmcount/libmcount-single.so: $(LIBMCOUNT_SINGLE_OBJS) $(LIBMCOUNT_UTILS_OBJS) $(LIBMCOUNT_ARCH_OBJS)
-	$(QUIET_LINK)$(CC) -shared -o $@ $^ $(LIB_LDFLAGS)
+$(objdir)/libmcount/libmcount-single.a: $(LIBMCOUNT_SINGLE_OBJS) $(LIBMCOUNT_UTILS_OBJS) $(LIBMCOUNT_ARCH_OBJS)
+	$(QUIET_LINK)ar rcs $@ $^
 
-$(objdir)/libmcount/libmcount-fast-single.so: $(LIBMCOUNT_FAST_SINGLE_OBJS) $(LIBMCOUNT_UTILS_OBJS) $(LIBMCOUNT_ARCH_OBJS)
-	$(QUIET_LINK)$(CC) -shared -o $@ $^ $(LIB_LDFLAGS)
+$(objdir)/libmcount/libmcount-fast-single.a: $(LIBMCOUNT_FAST_SINGLE_OBJS) $(LIBMCOUNT_UTILS_OBJS) $(LIBMCOUNT_ARCH_OBJS)
+	$(QUIET_LINK)ar rcs $@ $^
 
-$(objdir)/libmcount/libmcount-nop.so: $(LIBMCOUNT_NOP_OBJS)
-	$(QUIET_LINK)$(CC) -shared -o $@ $^ $(LIB_LDFLAGS)
+$(objdir)/libmcount/libmcount-nop.a: $(LIBMCOUNT_NOP_OBJS)
+	$(QUIET_LINK)ar rcs $@ $^
 
 $(LIBMCOUNT_ARCH_OBJS): $(wildcard $(srcdir)/arch/$(ARCH)/*.[cS]) $(LIBMCOUNT_DEPS)
 	@$(MAKE) -B -C $(srcdir)/arch/$(ARCH) $@
@@ -351,10 +351,10 @@ ifneq ($(findstring HAVE_LIBPYTHON, $(COMMON_CFLAGS)), )
 $(PYTHON_OBJS): $(objdir)/%.op: $(srcdir)/%.c $(COMMON_DEPS)
 	$(QUIET_CC_FPIC)$(CC) $(PYTHON_CFLAGS) -c -o $@ $<
 
-$(objdir)/python/uftrace_python.so: $(PYTHON_OBJS)
+$(objdir)/python/uftrace_python.a: $(PYTHON_OBJS)
 	$(QUIET_LINK)$(CC) -shared $(PYTHON_CFLAGS) -o $@ $(PYTHON_OBJS) $(PYTHON_LDFLAGS)
 else
-$(objdir)/python/uftrace_python.so:
+$(objdir)/python/uftrace_python.a:
 endif
 
 install: all
@@ -371,15 +371,15 @@ endif
 	$(call QUIET_INSTALL, uftrace)
 	$(Q)$(INSTALL) $(objdir)/uftrace         $(DESTDIR)$(bindir)/uftrace
 	$(call QUIET_INSTALL, libmcount)
-	$(Q)$(INSTALL) $(objdir)/libmcount/libmcount.so   $(DESTDIR)$(libdir)/libmcount.so
-	$(Q)$(INSTALL) $(objdir)/libmcount/libmcount-nop.so $(DESTDIR)$(libdir)/libmcount-nop.so
-	$(Q)$(INSTALL) $(objdir)/libmcount/libmcount-fast.so $(DESTDIR)$(libdir)/libmcount-fast.so
-	$(Q)$(INSTALL) $(objdir)/libmcount/libmcount-single.so $(DESTDIR)$(libdir)/libmcount-single.so
-	$(Q)$(INSTALL) $(objdir)/libmcount/libmcount-fast-single.so $(DESTDIR)$(libdir)/libmcount-fast-single.so
+	$(Q)$(INSTALL) $(objdir)/libmcount/libmcount.a   $(DESTDIR)$(libdir)/libmcount.a
+	$(Q)$(INSTALL) $(objdir)/libmcount/libmcount-nop.a $(DESTDIR)$(libdir)/libmcount-nop.a
+	$(Q)$(INSTALL) $(objdir)/libmcount/libmcount-fast.a $(DESTDIR)$(libdir)/libmcount-fast.a
+	$(Q)$(INSTALL) $(objdir)/libmcount/libmcount-single.a $(DESTDIR)$(libdir)/libmcount-single.a
+	$(Q)$(INSTALL) $(objdir)/libmcount/libmcount-fast-single.a $(DESTDIR)$(libdir)/libmcount-fast-single.a
 ifneq ($(findstring HAVE_LIBPYTHON, $(COMMON_CFLAGS)), )
 	$(call QUIET_INSTALL, uftrace-python)
 	$(Q)$(INSTALL) $(srcdir)/python/uftrace.py  $(DESTDIR)$(libdir)/uftrace.py
-	$(Q)$(INSTALL) $(objdir)/python/uftrace_python.so  $(DESTDIR)$(libdir)/uftrace_python.so
+	$(Q)$(INSTALL) $(objdir)/python/uftrace_python.a  $(DESTDIR)$(libdir)/uftrace_python.a
 endif
 	$(call QUIET_INSTALL, bash-completion)
 	$(Q)$(INSTALL) -m 644 $(srcdir)/misc/bash-completion.sh $(DESTDIR)$(completiondir)/uftrace
@@ -390,19 +390,19 @@ uninstall:
 	$(call QUIET_UNINSTALL, uftrace)
 	$(Q)$(RM) $(DESTDIR)$(bindir)/uftrace
 	$(call QUIET_UNINSTALL, libmcount)
-	$(Q)$(RM) $(DESTDIR)$(libdir)/libmcount.so
+	$(Q)$(RM) $(DESTDIR)$(libdir)/libmcount.a
 	$(call QUIET_UNINSTALL, libmcount-nop)
-	$(Q)$(RM) $(DESTDIR)$(libdir)/libmcount-nop.so
+	$(Q)$(RM) $(DESTDIR)$(libdir)/libmcount-nop.a
 	$(call QUIET_UNINSTALL, libmcount-fast)
-	$(Q)$(RM) $(DESTDIR)$(libdir)/libmcount-fast.so
+	$(Q)$(RM) $(DESTDIR)$(libdir)/libmcount-fast.a
 	$(call QUIET_UNINSTALL, libmcount-single)
-	$(Q)$(RM) $(DESTDIR)$(libdir)/libmcount-single.so
+	$(Q)$(RM) $(DESTDIR)$(libdir)/libmcount-single.a
 	$(call QUIET_UNINSTALL, libmcount-fast-single)
-	$(Q)$(RM) $(DESTDIR)$(libdir)/libmcount-fast-single.so
+	$(Q)$(RM) $(DESTDIR)$(libdir)/libmcount-fast-single.a
 ifneq ($(findstring HAVE_LIBPYTHON, $(COMMON_CFLAGS)), )
 	$(call QUIET_UNINSTALL, uftrace-python)
 	$(Q)$(RM) $(DESTDIR)$(libdir)/uftrace.py
-	$(Q)$(RM) $(DESTDIR)$(libdir)/uftrace_python.so
+	$(Q)$(RM) $(DESTDIR)$(libdir)/uftrace_python.a
 endif
 	$(call QUIET_UNINSTALL, bash-completion)
 	$(Q)$(RM) $(DESTDIR)$(completiondir)/uftrace
@@ -433,7 +433,7 @@ doc:
 
 clean:
 	$(call QUIET_CLEAN, uftrace)
-	$(Q)$(RM) $(objdir)/*.o $(objdir)/*.op $(objdir)/*.so $(objdir)/*.a
+	$(Q)$(RM) $(objdir)/*.o $(objdir)/*.op $(objdir)/*.a
 	$(Q)$(RM) $(objdir)/cmds/*.o $(objdir)/utils/*.o $(objdir)/misc/*.o
 	$(Q)$(RM) $(objdir)/utils/*.op $(objdir)/libmcount/*.op $(objdir)/python/*.op
 	$(Q)$(RM) $(objdir)/gmon.out $(srcdir)/scripts/*.pyc $(TARGETS)
